app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -386,10 +386,13 @@ def remove_middleware_datasource(ems_ref)
     end
 
     def emit_middleware_notification(type, op_name, op_arg, ems_ref)
+      mw_deployment = MiddlewareDeployment.find_by_ems_ref(ems_ref)
+      mw_server = mw_deployment.nil? ? MiddlewareServer.find_by_ems_ref(ems_ref) :
```

Comment:
@agrare I think we can get it from the specific methods, at some level, but where the operation is actually called (and the callback defined) is typically in a generic operation where this context is mostly lost and all we have is, again, the ems_ref.